CrimeMonitor/ConfigurationWindow.xaml.cs

```diff
@@ -0,0 +1,206 @@
+            crimeMonitor()?.writeRecord();
+        }
+
+        private void removeRecord(object sender, RoutedEventArgs e)
+        {
+            FactionRecord record = (FactionRecord)((Button)e.Source).DataContext;
+            crimeMonitor()?._RemoveRecord(record);
+            crimeMonitor()?.writeRecord();
+        }
+
+        private void updateRecord(object sender, RoutedEventArgs e)
+        {
+            FactionRecord record = (FactionRecord)((Button)e.Source).DataContext;
+            if (record.faction != Properties.CrimeMonitor.blank_faction)
+            {
+                Button updateButton = (Button)sender;
+                updateButton.Foreground = Brushes.Red;
+                updateButton.FontWeight = FontWeights.Bold;
+
+                Thread factionStationThread = new Thread(() =>
+                {
+                    crimeMonitor()?.GetFactionData(record, record.system);
+                    Dispatcher?.Invoke(() =>
+                    {
+                        updateButton.Foreground = Brushes.Black;
+                        updateButton.FontWeight = FontWeights.Regular;
+                    });
+                    crimeMonitor()?.writeRecord();
+                })
+                {
+                    IsBackground = true
+                };
+                factionStationThread.Start();
+            }
+        }
+
+        private void findIFRoute(object sender, RoutedEventArgs e)
+        {
+            Button updateButton = (Button)sender;
+            updateButton.Foreground = Brushes.Red;
+            updateButton.FontWeight = FontWeights.Bold;
+
+            Thread IFRouteThread = new Thread(() =>
+            {
+                string IFSystem = Navigation.Instance.GetFacilitatorRoute();
+                Dispatcher?.Invoke(() =>
+                {
+                    updateButton.Foreground = Brushes.Black;
+                    updateButton.FontWeight = FontWeights.Regular;
+
+                    // If 'next system' found, send to clipboard
+                    if (IFSystem != null)
+                    {
+                        Clipboard.SetData(DataFormats.Text, IFSystem);
+                    }
+                });
+            })
+            {
+                IsBackground = true
+            };
+            IFRouteThread.Start();
+        }
+
+        private void profitShareChanged(object sender, TextChangedEventArgs e)
+        {
+            CrimeMonitorConfiguration configuration = CrimeMonitorConfiguration.FromFile();
+            try
+            {
+                int? profitShare = string.IsNullOrWhiteSpace(crimeProfitShareInt.Text) ? 0 : Convert.ToInt32(crimeProfitShareInt.Text, CultureInfo.InvariantCulture);
+                crimeMonitor().profitShare = profitShare;
+                configuration.profitShare = profitShare;
+                configuration.ToFile();
+            }
+            catch
+            {
+                // Bad user input; ignore it
```

Comment:
Should we delete the bad user input from the UI or somehow signal in the UI that the input isn't acceptable?

Reply:
Also, what happens if a user enters a negative value, a decimal value, or a value that doesn't lie between 0-100?

Reply:
Removed `profitshare` property and all associated code.